pkgs/tools/networking/pykms/default.nix

```diff
@@ -29,6 +29,29 @@ let
     fi
   '');
 
+  loggingPatch = 
+    (builtins.toFile "log-to-current-directory-by-default.patch" ''
+# By default, create log files in current directory.
+--- ../original/py-kms/pykms_Client.py
++++ py-kms/pykms_Client.py
+@@ -48,5 +48,5 @@
+                     'choi' : ["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG", "MINI"]},
+         'lfile' : {'help' : 'Use this option to set an output log file. The default is \"pykms_logclient.log\" or type \"STDOUT\" to view log info on stdout.',
+-                   'def' : os.path.dirname(os.path.abspath( __file__ )) + "/pykms_logclient.log", 'des' : "logfile"},
```

Comment:
Did you approach upstream on that? The help text reads like it should log to `pykms_logclient.log` in the current working directory, not relative to the location on where `pykms` is installed.

This should cause problems in other distros as well, if `pykms` is installed by the distros package manager, and not world-writeable.

Reply:
Well, creating the log file in the executable's directory seems to be a deliberate decision made upstream. I don't think I'm willing to debate it with them.

Reply:
I opened https://github.com/SystemRage/py-kms/issues/64 to discuss this upstream.

Reply:
@flokli Thank you.

Reply:
Do we really care about this? pykms can be given an absolute path to log to (or STDOUT which is what the the module currently does happening).

Reply:
@peterhoeg I think this is worth caring for. It's better if the program works out-of-the-box, without requiring to specify this kind of parameters.

At the very least, I firmly disagree with the way the logging behaviour is tuned by the nixpkgs package right now.